scripts/mujoco/create_dataset_humanoid_standup.py

```diff
@@ -0,0 +1,142 @@
+
+    def __call__(self, env, **kwargs):
+        step_data = super().__call__(env, **kwargs)
+        if getattr(env.unwrapped, "_include_cinert_in_observation", True) is False:
+            step_data["observation"] = np.concatenate(
+                [step_data["observation"], env.unwrapped.data.cinert[1:].flatten()]
+            )
+        if getattr(env.unwrapped, "_include_cvel_in_observation", True) is False:
+            step_data["observation"] = np.concatenate(
+                [step_data["observation"], env.unwrapped.data.cvel[1:].flatten()]
+            )
+        if getattr(env.unwrapped, "_include_qfrc_actuator_in_observation", True) is False:
+            step_data["observation"] = np.concatenate(
+                [step_data["observation"], env.unwrapped.data.qfrc_actuator[6:].flatten()]
+            )
+        if getattr(env.unwrapped, "_include_cfrc_ext_in_observation", True) is False:
+            step_data["observation"] = np.concatenate(
+                [step_data["observation"], env.unwrapped.data.cfrc_ext[1:].flat.copy()]
+            )
+
+        return step_data
+
+
+def create_dataset_from_policy(dataset_id, collector_env, policy, n_steps: int, algorithm_name):
+    truncated = True
+    terminated = True
+    seed = 123
+    for step in tqdm(range(n_steps)):
+        if terminated or truncated:
+            obs, _ = env.reset(seed=seed)
+            seed += 1
+            if (n_steps - step) < collector_env.spec.max_episode_steps :  # trim trailing non-full episodes
+                break
+
+        action = policy(obs)
+        obs, _, terminated, truncated, _ = env.step(action)
+
+    return collector_env.create_dataset(
+        dataset_id=f"mujoco/{dataset_id}",
+        algorithm_name="SB3/{algorithm_name}",
+        code_permalink="https://github.com/Farama-Foundation/minari-dataset-generation-scripts",
+        author="Kallinteris Andreas",
+        author_email="kallinteris@protonmail.com",
+        requirements=[
+            "mujoco==3.2.3",
+        ],
+    )
+
+
+def load_policy(env_id: str, algo: str, proficiency: str):
+    model_checkpoint = load_from_hub(
+        repo_id=f"farama-minari/{env_id}-v5-{algo.upper()}-{proficiency}",
+        filename=f"{env_id.lower()}-v5-{algo.lower()}-{proficiency}.zip",
+    )
+
+    match algo:
+        case "SAC":
+            policy = SAC.load(model_checkpoint)
+        case "TD3":
+            policy = TD3.load(model_checkpoint)
+        case "PPO":
+            policy = PPO.load(model_checkpoint)
+
+    return policy
+
+
+def make_env(env_id: str):
+    """Wrapper to create the appropriate environment."""
+    if env_id == "HumanoidStandup":
+        env = gym.make(f"{env_id}-v5", include_cinert_in_observation=False, include_cvel_in_observation=False, include_qfrc_actuator_in_observation=False, include_cfrc_ext_in_observation=False)
+    else:
+        env = gym.make(f"{env_id}-v5", render_mode="rgb_array",)
+    return env
```

Comment:
this function and the one above looks general, but they are written only for a single env and algo, isn't it?

Reply:
Yes, I copied the functions from create data set and I added the requirements for humanoid stand-up.